homeassistant/components/media_player/sonos.py

```diff
@@ -255,10 +258,37 @@ def wrapper(*args, **kwargs):
             return funct(*args, **kwargs)
         except SoCoException as err:
             _LOGGER.error("Error on %s with %s", funct.__name__, err)
-
     return wrapper
 
 
+def soco_filter_upnperror(errorcodes=None):
+    """Filter out specified UPnP errors from logs."""
+    def decorator(funct):
+        """Decorator function."""
+        @ft.wraps(funct)
+        def wrapper(*args, **kwargs):
+            """Wrap for all soco UPnP exception."""
+            from soco.exceptions import SoCoUPnPException
+
+            # Temporarily disable SoCo logging because it will log the
+            # UPnP exception otherwise
+            _SOCO_SERVICES_LOGGER.disabled = True
+
+            returnval = None
+            try:
+                returnval = funct(*args, **kwargs)
+            except SoCoUPnPException as err:
+                if errorcodes is not None and err.error_code in errorcodes:
+                    pass
+                else:
+                    raise
+
+            _SOCO_SERVICES_LOGGER.disabled = False
```

Comment:
Add this to the `finally` block from try…catch so that it's guaranteed to enable logging after it runs.